keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensor.java

```diff
@@ -463,6 +470,7 @@ public DoubleTensor atan2InPlace(double y) {
     @Override
     public DoubleTensor atan2InPlace(DoubleTensor y) {
         if (y.isScalar()) {
+            Nd4j.setDataType(DataBuffer.Type.DOUBLE);
```

Comment:
What's this doing here?

Reply:
The ones that are created need to be a double buffer or things will break. I've factored out the creation of all INDArrays to a class that forces you to provide a buffer type instead of defaulting to what ND4J thinks. Also there's `tensor.rsubi(double x)` which when x is 1 is the same as saying `ones(..).subi(tensor)`. Using rsubi removes the need to do a new buffer creation.

Reply:
cool